packages/bs-reform/src/ReForm.re

```diff
@@ -93,22 +106,43 @@ module Make = (Config: Config) => {
       let Validation.Schema(validators) = schema;
       validators->Belt.Array.map(validator =>
         switch (validator) {
-        | Validation.IntMin(field, _min) => (
+        | Validation.IntMin(field, _min, _customErrorMessage) => (
```

Comment:
hmm custom messages could be passed via `i18n` dictionary

Reply:
I've looked through ReSchema.re and ReSchemaI18n.re and am not sure how you would accomplish this. Can you give me an example of what you're thinking?

Reply:
ofc! 
here's how:

```reason
let formErrorsDictionary: ReSchemaI18n.t = {
  ...ReSchemaI18n.default,
  stringMin: (~value as _value, ~min) =>
    "Este valor deve ter pelo menos " ++ string_of_int(min) ++ " caracteres",
  stringMax: (~value as _value, ~max) =>
    "Este valor deve ter no máximo " ++ string_of_int(max) ++ " caracteres",
};

[@react.component]
let make = (~onSubmit, ~navigation, ~initialState) => {
  module AppConfig = (val AppConfigTypes.useAppConfig());

  let snackbar = Snackbar.useSnackbar();

  let reform: ProfileFormHook.api =
    ProfileFormHook.use(
      ~i18n=formErrorsDictionary,
      ~schema=
        ProfileFormHook.Validation.Schema([|
```

Reply:
My initial impression when seeing this is that it doesn't allow for validation error messages on a per-field basis. While you could do something more general for errors like "This field is required", I also think there is a strong use case to have errors custom to that specific field like "A username is required." This is also helpful if it's a large form and a summary of errors is needed at the bottom of the form.

I haven't tried this but maybe another option would be to have a second "Custom" validator that allows you to pass in a default validator and a custom error message?

What are your thoughts here?

Reply:
@fakenickels Any thoughts here?

Reply:
>I haven't tried this but maybe another option would be to have a second "Custom" validator that allows you to pass in a default validator and a custom error message?

I don't know if I could get what you are trying to say, but isn't that how `Custom` works already?

Reply:
Right, so the difference is that Custom requires you put a custom validation and custom error message. What I'm suggesting is a way to use currently predefined validations but with custom error messages.